test/test_transformers.py

```diff
@@ -724,6 +731,58 @@ def rand_tensor(*shape):
             if dropout_p == 0.0 or device == 'cpu':
                 self.assertEqual(actual, expected)
 
+    @unittest.skipIf(TEST_WITH_CROSSREF, 'Fastpath not available with crossref')
+    def test_mask_check_fastpath(self):
+        """
+        Test that fastpath is executed independently of the mask that is passed.
+        If the passed mask is left aligned, test that nested tensors are used (sparsity fastpath),
+        otherwise use fastpath with traditional tensors.
+        """
+
+        with torch.no_grad():
+            x = torch.Tensor([[[1, 2], [3, 4], [5, 6]]]).to(torch.float)
+
+            def _test_fastpath(model, mask, mock_return_value, nested_tensors=True):
+                with patch('torch._transformer_encoder_layer_fwd') as fastpath_mock:
```

Comment:
`_transformer_encoder_layer_fwd` represents the "sparsity fastpath", correct? Maybe we should name the mock with `sparsity` to avoid ambiguity as to which fastpath is being mocked out

Reply:
Not exactly, `_transformer_encoder_layer_fwd` represents both the "sparsity fastpath" and normal fastpath. When the method is called with nested tensors, that is the sparsity fastpath, otherwise it is normal fastpath.

Reply:
Okay gotcha. How are we validating whether the "sparsity fastpath" or "normal fastpath" is taken? Is this based on the args passed to `_transformer_encoder_layer_fwd`?